packages/landing/src/components/debug.tsx

```diff
@@ -270,6 +309,48 @@ export class Debug extends Component<{ map: maplibregl.Map }, DebugState> {
     this.setTerrainShown(sourceId);
   };
 
+  selectHillShade = (event: React.ChangeEvent<HTMLSelectElement>): void => {
+    const sourceId = event.target.value;
+    this.setHillShadeShown(sourceId);
+  };
+
+  setHillShadeShown(sourceId: string | null): void {
+    Config.map.setDebug('debug.hillshade', sourceId);
+    const map = this.props.map;
+    const isTurnOff = sourceId === 'off' || sourceId == null;
+
+    const currentLayer = map.getLayer(HillShadeLayerId);
+    if (isTurnOff) {
+      if (currentLayer) map.removeLayer(HillShadeLayerId);
+      return;
+    }
+
+    if (currentLayer?.source === sourceId) return;
+
+    // Hillshading from an existing raster-dem source gives very mixed results and looks very blury
+    // so add a new source layer to generate from
+    const hillShadeSourceId = `${HillShadePrefix}${sourceId}`;
+    const existingSource = map.getSource(hillShadeSourceId);
+    if (existingSource == null) {
+      const source = map.getSource(sourceId);
+      if (source?.type !== 'raster-dem') {
+        // Source cannot be found, config is invalid
+        Config.map.setDebug('debug.hillshade', null);
+        return;
+      }
+
+      map.addSource(hillShadeSourceId, { ...source.serialize(), type: 'raster-dem', id: undefined });
```

Comment:
Why do we got a `id: undefined`?

Reply:
we need to force the id to be undefined.